nodes/src/nodes/tool_pipe/IGlobal.py

```
@@ -0,0 +1,70 @@
+
+    def beginGlobal(self) -> None:
+        if self.IEndpoint.endpoint.openMode == OPEN_MODE.CONFIG:
+            return
+
+        cfg = Config.getNodeConfig(self.glb.logicalType, self.glb.connConfig)
+
+        self.tool_description = str(cfg.get('tool_description') or '').strip()
+        self.return_type = str(cfg.get('return_type') or 'text').strip()
+
+        if self.return_type not in VALID_RETURN_TYPES:
+            raise Exception(f'tool_pipe: return_type must be one of {sorted(VALID_RETURN_TYPES)}')
```

Ruff (0.15.9)

[warning] Create your own exception

(TRY002)